source/customization.rst

```diff
@@ -1,32 +1,49 @@
 .. _customization:
 
-Customization
-=============
+Customization (or the many OOD environment variables)
+=====================================================
 
-Announcements
----------------
+Setting the PUN Environment With ``nginx_stage.yml``
+----------------------------------------------------
 
-.. attention:: TODO
+The configuration file ``nginx_stage.yml`` is the best way to set the environment which the PUN will use. Defining the mapping ``pun_custom_env`` allows setting Open OnDemand specific variables and is used for branding, message of the day setting, quota warnings, whitelists and other uses. For environment variables that OOD applications may need, are already in the environment and should be exposed to OOD (``$PATH``, ``$LD_LIBRARY_PATH``, etc...) define the sequence ``pun_custom_env_declarations``.
```

Comment:
It is still valuable at this stage to be able to set some per-app configuration in the individual app's dotenv files. In fact I think most of the configuration variables at this stage should be still set in the apps themselves and not in nginx_stage.yml. I don't think it is appropriate for message of the day or quota warnings. I think it is appropriate for the whitelist and branding since these are intended to be used across all apps. The motd I would keep with the dashboard config for now. I think its actually a negative that once the PUN has it, 100% of the apps have it set, regardless if whether you are running an app in production mode, or development mode, whether it is a core app or a third party app. And default dotenv functionality would not override it. So I think we be careful what we encourage to be set here.

This is at least what Jeremy and I have discussed in the passed. Though feel free to push back and we can discuss. In the future I would imagine we would design a better centralized OnDemand configuration that all apps would have access to, that would not be in the environment. And branding might be delegated to a css file.